nemo/collections/asr/parts/mixins/mixins.py

```diff
@@ -112,43 +200,47 @@ def get_vocab():
 
         else:
             # This is a WPE Tokenizer
             # If path from previous registration exists, remove it
-            if 'vocab_path' in self.tokenizer_cfg:
-                vocab_path = self.tokenizer_cfg.get('vocab_path')
+            if 'vocab_path' in tokenizer_cfg:
+                vocab_path = tokenizer_cfg.get('vocab_path')
+            else:
+                vocab_path = os.path.join(tokenizer_dir, 'vocab.txt')
+
+            if lang is not None:
+                vocab_path = self.register_artifact('tokenizer.tokenizers.' + lang + '.vocab_path', vocab_path)
             else:
-                vocab_path = os.path.join(self.tokenizer_dir, 'vocab.txt')
-            vocab_path = self.register_artifact('tokenizer.vocab_path', vocab_path)
-            self.vocab_path = vocab_path
+                vocab_path = self.register_artifact('tokenizer.vocab_path', vocab_path)
 
             # If path from previous registration exists, remove it
-            if 'vocab_path' in self.tokenizer_cfg:
-                self.tokenizer_cfg.pop('vocab_path')
+            if 'vocab_path' in tokenizer_cfg:
+                tokenizer_cfg.pop('vocab_path')
 
-            self.tokenizer = tokenizers.AutoTokenizer(
+            hf_tokenizer_kwargs = tokenizer_cfg.get('hf_kwargs', {})
+            tokenizer = tokenizers.AutoTokenizer(
                 pretrained_model_name='bert-base-cased',
-                vocab_file=self.vocab_path,
-                mask_token=self.hf_tokenizer_kwargs.get('mask_token', None),
-                bos_token=self.hf_tokenizer_kwargs.get('bos_token', None),
-                eos_token=self.hf_tokenizer_kwargs.get('eos_token', None),
-                pad_token=self.hf_tokenizer_kwargs.get('pad_token', None),
-                sep_token=self.hf_tokenizer_kwargs.get('sep_token', None),
-                cls_token=self.hf_tokenizer_kwargs.get('cls_token', None),
-                unk_token=self.hf_tokenizer_kwargs.get('unk_token', None),
-                use_fast=self.hf_tokenizer_kwargs.get('use_fast', False),
+                vocab_file=vocab_path,
+                mask_token=hf_tokenizer_kwargs.get('mask_token', None),
+                bos_token=hf_tokenizer_kwargs.get('bos_token', None),
+                eos_token=hf_tokenizer_kwargs.get('eos_token', None),
+                pad_token=hf_tokenizer_kwargs.get('pad_token', None),
+                sep_token=hf_tokenizer_kwargs.get('sep_token', None),
+                cls_token=hf_tokenizer_kwargs.get('cls_token', None),
+                unk_token=hf_tokenizer_kwargs.get('unk_token', None),
+                use_fast=hf_tokenizer_kwargs.get('use_fast', False),
             )
 
         logging.info(
-            "Tokenizer {} initialized with {} tokens".format(
-                self.tokenizer.__class__.__name__, self.tokenizer.vocab_size
-            )
+            'Tokenizer {} initialized with {} tokens'.format(tokenizer.__class__.__name__, tokenizer.vocab_size)
```

Comment:
Missed multiple assignments.

Reply:
This one does not assign an instance variable.. it just makes a tokenizer and returns..